src/App.js

```diff
@@ -0,0 +1,316 @@
+        <HomeSearchPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          onChangePage={handleChangePage}
+          onChangeList={handleChangeList}
+          onChangeTask={handleChangeTask}
+          onDeleteList={handleDeleteList}
+          onToggleDeleteAlert={handleToggleDeleteAlert}
+          showDeleteAlert={showDeleteAlert}
+        />
+      ) : null}
+      {currentPage === "SingleListPage" ? (
+        <SingleListPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          currentPage={currentPage}
+          onChangePage={handleChangePage}
+          onChangeTask={handleChangeTask}
+          onEditList={handleEditList}
+          onEditTask={handleEditTask}
+          onDeleteCompleted={handleDeleteCompletedTasks}
+          onDeleteAllTasks={handleDeleteAllTasks}
+          onDeleteList={handleDeleteList}
+          onCreateTask={handleChangeTask}
+          onToggleDeleteAlert={handleToggleDeleteAlert}
+        />
+      ) : null}
+      {currentPage === "ListSearchPage" ? (
+        <ListSearchPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          onChangePage={handleChangePage}
+          onChangeTask={handleChangeTask}
+        />
+      ) : null}
+      {currentPage === "ViewTaskPage" ? (
+        <ViewEditCreateTaskPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          onChangePage={handleChangePage}
+          inEditTaskMode={false}
+          inCreateTaskMode={false}
+        />
+      ) : null}
+      {currentPage === "EditTaskPage" ? (
+        <ViewEditCreateTaskPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          onChangePage={handleChangePage}
+          onCreateTask={handleCreateTask}
+          onDeleteTask={handleDeleteTask}
+          onEditAllTaskFields={handleEditTaskAllFields}
+          inEditTaskMode={true}
+          inCreateTaskMode={false}
+          onToggleDeleteAlert={handleToggleDeleteAlert}
+          showDeleteAlert={showDeleteAlert}
+        />
+      ) : null}
+      {currentPage === "CreateTaskPage" ? (
+        <ViewEditCreateTaskPage
+          data={data}
+          prevPage={prevPage}
+          currentListId={currentListId}
+          currentTaskId={currentTaskId}
+          onChangePage={handleChangePage}
+          onDeleteTask={handleDeleteTask}
+          onCreateTask={handleCreateTask}
+          onEditAllTaskFields={handleEditTaskAllFields}
+          inEditTaskMode={false}
+          inCreateTaskMode={true}
+        />
+      ) : null}
+      {currentPage === "EditListPage" ? (
```

Comment:
This EditCreateListPage and the next are extremely similar (difference is inEditListMode and inCreateListMode).  Would be nicer to combine into one <EditCreateListPage where you set those two attributes based on currentPage.